Change default open to io.open to handle unicode in freebsd-version

pludi opened the pull request:
The file `bin/freebsd-version` in each Jail contains a maintainer-line, which for 12.0 is
```
# Copyright (c) 2013 Dag-Erling Smørgrav
```
which contains a Unicode character that trips the `startswith` in https://github.com/iocage/iocage/blob/1e5339376f376945c365c7d2e5be045aee4b0ff9/iocage_lib/ioc_common.py#L898-L902

This fix changes `open` to `io.open` including setting the correct character set, allowing the code to finish without exception.

pludi committed:
Change default open to io.open to handle unicode in freebsd-version

pludi committed:
Fix line-length

skarekrow commented:
Thanks for this! There's more usages of `freebsd-version` in the program (including some variables named `freebsd_version`), so we should tackle all of those

skarekrow commented:
Also as far as I know, `io.open` is the same as `open`, so we can drop an import line and just specify the encoding in the open statement?

pludi committed:
*   Replace `io.open` with `open`
*   Replace additional occurence in ioc_json.py

pludi commented:
> Also as far as I know, `io.open` is the same as `open`, so we can drop an import line and just specify the encoding in the open statement?

True. Changed that.

> There's more usages of freebsd-version in the program (including some variables named freebsd_version), so we should tackle all of those

I found one other in `ioc_json.py`. The others are checked either through `ioc_common.get_jail_freebsd_version` (which I changed) or `ioc_common.checkoutput`, which already decodes to UTF-8.

pludi committed:
Remove io import

sonicaj reviewed (commented):
You missed two usages in
1)https://github.com/iocage/iocage/blob/a09857be4f3ef411d0ffd08d4b49fb7267ae7cf8/iocage_lib/ioc_plugin.py#L308
2) https://github.com/iocage/iocage/blob/a09857be4f3ef411d0ffd08d4b49fb7267ae7cf8/iocage_lib/ioc_plugin.py#L327

Can you please have them covered as well ? Thank you ; )

pludi committed:
Merge branch 'master' into fix-update-unicode-error

pludi committed:
Fix 2 unicode-open in ioc_plugin.py

pludi commented:
True. Changed them & merged current `master` as well.

sonicaj reviewed (approved):
Perfect, thank you! Let's wait for the tests ; )

pludi commented:
Tests are green.

sonicaj commented:
Thank you ; )
